lib/logstash/filters/ruby.rb

```diff
@@ -66,6 +66,11 @@ def register
     else
       @logger.fatal("You must either use an inline script with the \"code\" option or a script file using \"path\".")
     end
+  rescue SyntaxError => e
+    @logger.error("Syntax Error in register #{e.message}",
+                  :class     => e.class.name,
+                  :backtrace => e.backtrace)
+    fail("Ruby script syntax error")
```

Comment:
might be better to handle `ScriptError` in general (covers `ScriptError` as well as `LoadError` etc.) e.g.
```suggestion
  rescue ScriptError => e
    @logger.error("error in register", message: e.message, exception: e.class, backtrace: e.backtrace)
    fail("unexpected error: #{e}")
```
... just a suggestion I did not test this - might require more refining

Reply:
`ScriptError` is a `StandardError`. So, rescue ScriptError and SyntaxError